src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.spec.ts

```diff
@@ -1,30 +1,247 @@
+  let statusMessageService: StatusMessageService;
+  let simpleModalService: SimpleModalService;
+  let timezoneService: TimezoneService;
+
+  const notificationTableRowModel1: NotificationsTableRowModel = {
+    isHighlighted: true,
+    notification: EXAMPLE_NOTIFICATION_ONE,
+  };
+
+  const notificationTableRowModel2: NotificationsTableRowModel = {
+    isHighlighted: false,
+    notification: EXAMPLE_NOTIFICATION_TWO,
+  };
+
+  /**
+   * Verifies the row model is updated as expected.
+   */
+  function verifyFirstRowModelEqualToExample(expected: Notification): void {
+    expect(component.notificationsTableRowModels[0].notification.notificationId).toEqual(expected.notificationId);
+    expect(component.notificationsTableRowModels[0].notification.shown).toBeFalsy();
+    expect(component.notificationsTableRowModels[0].notification.targetUser).toEqual(expected.targetUser);
+    expect(component.notificationsTableRowModels[0].notification.title).toEqual(expected.title);
+    expect(component.notificationsTableRowModels[0].notification.message).toEqual(expected.message);
+  }
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
       imports: [
         AdminNotificationsPageModule,
         HttpClientTestingModule,
+        BrowserAnimationsModule,
+      ],
+      providers: [
+        TimezoneService,
       ],
     })
     .compileComponents();
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(AdminNotificationsPageComponent);
+    notificationService = TestBed.inject(NotificationService);
+    statusMessageService = TestBed.inject(StatusMessageService);
+    simpleModalService = TestBed.inject(SimpleModalService);
+    timezoneService = TestBed.inject(TimezoneService);
     component = fixture.componentInstance;
+    moment.tz.setDefault('UTC');
+    jest.spyOn(timezoneService, 'guessTimezone').mockReturnValue('Asia/Singapore');
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default view', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when notifications are loading', () => {
+    component.isNotificationLoading = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when notifications failed to load', () => {
+    component.hasNotificationLoadingFailed = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when notification edit form expanded for adding notification', () => {
+    jest.spyOn(notificationService, 'getNotifications').mockReturnValue(of());
+    component.initNotificationEditFormModel();
+    component.loadNotifications();
+    fixture.detectChanges();
+
+    const btn: any = fixture.debugElement.nativeElement.querySelector('#btn-add-notification');
+    expect(btn.disabled).toBeFalsy();
+    expect(btn.textContent).toEqual(' Add New Notification');
```

Comment:
Why is there a space before the words `Add New Notification` for the button? Was it there to add spacing?

Reply:
Yes, I believe it follows the format of `Add Courses` button in `instructor-home-page.component.html`

Reply:
Seems that it is not recommended to include date time and use large snapshot, so I will remove the `toMatchSnapshot` for this test and use test on smaller elements such as whether the edit form is expanded from variables.